### DISCUSSION TITLE
`when` clause on `default` label

### DISCUSSION THREAD

**Discussion by kyoyama-kazusa** [dotnet/csharplang#7679] [Language Ideas]
There is no support for `when` clause followed with a `default` label inside a `switch` statement:

```csharp
switch (expr)
{
    default when ...:
        break;
}
```

I must write `case var _ when ...` instead:

```csharp
switch (expr)
{
    case var _ when ...:
        break;
}
```

Because a discard cannot be used as a top-level pattern, I cannot use simpler syntax like `case _ when ...`.

Any suggestions here? Thanks.

**Comment by KennethHoff** [comment]
This feels like.. I don't know if oversight is the best word, but certainly _feels_ inconsistent with switch expressions, because the following is legal:

```c#
var value = expr switch
{
	_ when a => ... ,
	_ when b => ... ,
	_ when c => ... ,
}
```

Granted, I guess it technically is the same syntax as the alternative mentioned above (`case var _ when a: ...`), but it's not idiomatic like how it would be with switch exprs.

**Comment by 333fred** [comment]
I'd be willing to champion `default when` for switch statements.
